pkg/cli/codearea.go

```diff
@@ -207,9 +261,19 @@ func (w *codeArea) handleKeyEvent(key ui.Key) bool {
 		}
 		return true
 	}
+
+	// Perform small word abbreviation expansion if a non-small word char is
+	// typed. This mimics how abbreviations work in some editors. This needs
+	// to be done before the overlay handler test to ensure it happens on
+	// newline being typed.
```

Comment:
I'd rather expansions to *not* to be done when a newline is typed. Expansions are supposed to give users real-time feedback, giving them a chance to correct the expansion if that's not what they need. Expanding an abbreviation when the user types a newline means that the user never gets a chance to actually see what got expanded.

I'd rather live with (and force users to live with) the slight inconvenience of having to say, type a space, to see the expansion.

Reply:
Okay, I can live with that. But note that the "expand on newline" behavior is how fish has behaved for many years and it doesn't appear to cause problems for users. I can't recall anyone opening an issue or raising this as a problem on the fish IM channels.